File: problems-adapt/1901-2000/1967_patterns-embedded-in-the-word/solution.java
class Solution {

    public int countEmbeddedPatterns(String[] patterns, String word) {
        // Each pattern is judged on its own: count the ones that occur as
        // a contiguous substring of word.
        int count = 0;
        for (String pattern : patterns) {
            if (word.contains(pattern)) {
                ++count;
            }
        }
        return count;
    }
}

File: problems-adapt/1901-2000/1967_patterns-embedded-in-the-word/solution.js
/**
 * @param {string[]} patterns
 * @param {string} word
 * @return {number}
 */
var countEmbeddedPatterns = function (patterns, word) {
    // Each pattern is judged on its own: count the ones that occur as a
    // contiguous substring of word.
    let count = 0;
    for (const pattern of patterns) {
        if (word.includes(pattern)) {
            ++count;
        }
    }
    return count;
};
